addons/point_of_sale/models/pos_session.py

```diff
@@ -345,6 +351,20 @@ def open_cashbox(self):
 
         return action
 
+    @api.multi
+    def action_view_order(self):
+        action = {
+            'name': 'Orders',
+            'res_model': 'pos.order',
+            'view_mode': 'tree,form',
+            'type': 'ir.actions.act_window',
+            'domain': [('session_id', 'in', self.ids)],
+        }
+        if len(self.order_ids) == 1:
```

Comment:
I wouldn't handle that case; if you have only one order in your session it doesn't shock me to have the list view open anyway...
this makes sense when it's frequent for an o2m field to lead to a single record (e.g. link between SO and invoice: you expect only one invoice per SO most of the time); less so when you expect the normal case to be 'more than one'

Reply:
Resolved